contrib/udfs/src/main/java/org/apache/drill/exec/udfs/DateFunctions.java

```diff
@@ -140,8 +143,77 @@ public void eval() {
       java.time.format.DateTimeFormatter formatter = java.time.format.DateTimeFormatter.ofPattern(format);
       java.time.LocalDateTime dateTime = java.time.LocalDateTime.parse(inputDate, formatter);
 
-      java.time.LocalDateTime td = org.apache.drill.exec.udfs.NearestDateUtils.getDate(dateTime, intervalString);
+      java.time.LocalDateTime td = DateConversionUtils.getDate(dateTime, intervalString);
       out.value = td.atZone(java.time.ZoneId.of("UTC")).toInstant().toEpochMilli();
     }
   }
+
+  @FunctionTemplate(names = {"yearweek","year_week"},
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class YearWeekFunction implements DrillSimpleFunc {
+    @Param
+    VarCharHolder inputHolder;
+
+    @Output
+    IntHolder out;
+
+    @Override
+    public void setup() {
+      // noop
+    }
+
+    @Override
+    public void eval() {
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputHolder.start, inputHolder.end, inputHolder.buffer);
+      java.time.LocalDateTime dt = org.apache.drill.exec.udfs.DateUtilFunctions.getTimestampFromString(input);
+      int week = dt.get(java.time.temporal.IsoFields.WEEK_OF_WEEK_BASED_YEAR);
+      int year = dt.getYear();
+      out.value = (year * 100) + week;
+    }
+  }
+
+  @FunctionTemplate(names = {"yearweek","year_week"},
```

Comment:
I admit that I don't know all of the detail but I understand that because year_week(x) = 100*year(x) + week(x) we have the opportunity to provide "no code" implementation of this function in terms of existing functions by making an addition to DrillConvertletTable.

Reply:
@jnturton We don't have `year()` and `week()` functions unfortunately.  I do plan on adding those as well as `quarter()` and a few others.  It is possible now to do some of that but the syntax is really non-intuitive.

Reply:
Basically, what I was hoping to do was follow this as a guide: https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html.

Reply:
> @jnturton We don't have `year()` and `week()` functions unfortunately. I do plan on adding those as well as `quarter()` and a few others. It is possible now to do some of that but the syntax is really non-intuitive.

They work for me...

```
apache drill> select 100*year(current_date)+week(current_date);
EXPR$0  202243

1 row selected (0.562 seconds)
```

Reply:
Wow... I hate it when good functionality is not documented...  Thanks for letting me know about this and I've since updated the Drill docs. (https://github.com/apache/drill-site/pull/34)